File: Python/8py 3/remove.py
"""
|--------------------------------------------------------------------------
| Exclamation marks series #1: Remove an exclamation mark from the end of string
|--------------------------------------------------------------------------
|
"""

def remove(s):
   return s if len(s) == 0 or s[-1] != '!' else s[slice(0, -1)]


def remove(s):
   return s[:-1] if s.endswith('!') else s


from re import sub
def remove(s):
   return sub(r'!$', '', s)


def remove(s):
   return s[:-1] if s and s[-1] == '!' else s


print(remove("Hi!"), "Hi")
print(remove("Hi!!!"), "Hi!!")
print(remove("!Hi"), "!Hi")
print(remove("!Hi!"), "!Hi")
print(remove("Hi! Hi!"), "Hi! Hi")
print(remove("Hi"), "Hi")
print(remove(""), "")

File: JavaScript/8js 2/remove.js
/*
|--------------------------------------------------------------------------
| Exclamation marks series #1: Remove an exclamation mark from the end of string
|--------------------------------------------------------------------------
|
*/

function remove(string) {
   return string[string.length - 1] == '!' ? string.slice(0, string.length - 1) : string;
}


const remove = s => s.replace(/!$/, '');


function remove(s) {
   return s.endsWith('!') ? s.slice(0, -1) : s;
}


function remove(s) {
   return s.charAt(s.length - 1) == "!" ? s.substr(0, s.length - 1) : s;
}


console.log(remove("Hi!"), "Hi");
console.log(remove("Hi!!!"), "Hi!!");
console.log(remove("!Hi"), "!Hi");
console.log(remove("!Hi!"), "!Hi");
console.log(remove("Hi! Hi!"), "Hi! Hi");
console.log(remove("Hi"), "Hi");